llama/runner/runner.go

```diff
@@ -211,158 +223,159 @@ func (s *Server) run(ctx context.Context) {
 		case <-ctx.Done():
 			return
 		default:
-			slog.Debug("Processing batch", "seqs", len(s.seqs))
-			s.mu.Lock()
-			for s.allNil() {
-				s.cond.Wait() // Wait until an item is added
-			}
-			s.mu.Unlock()
+			pieces = s.processBatch(pieces)
+		}
+	}
+}
 
-			for i, seq := range s.seqs {
-				if seq == nil {
-					continue
-				}
+func (s *Server) processBatch(pieces [][]string) [][]string {
```

Comment:
Passing a `[][]string` around here might be a bit brittle, and I was never a big fan of the `pieces` name. Should we maybe keep track of the response-so-far in `seq.response` or similar?

Reply:
That's a good suggestion - I also wasn't that excited about how `pieces` was being passed around.